### PR TITLE
Consolidate student search methods

### LINKED ISSUE #84: Consolidate student search methods
There are multiple routes and functions that handle searching for students.
`app/logic/searchVolunteers.py`, `app/logic/searchStudents.py`, and ` searchVolunteers() in app/controllers/admin/routes.py` all have a different implementation.

We need to consolidate to a single search method. multiple routes are ok, as long as they are for different purposes and handled by (probably) one controller method.

### PR DESCRIPTION
**Pull request by mupotsal**
Issue #84 : The logic for Searching students and Volunteers was in two different files but the functions within them had little differences. The main difference was the type of the data they were returning. 

Solution: Combined the two functions into one called searchVolunteersAndStudents in searchVolunteersAndStudents.py

Test:

1. Navigate to /1/1/track_volunteers
2. Click the + sign to get the search modal pop up. 
3. Begin Typing "Zach" and a student named Zach should pop up. 
4. Testing for other places where the student is searched is still in progress since the search function have been used in different branch

Screen Shot 2021-08-31 at 10 45 21 AM

### CODE REVIEW

File: app/logic/searchVolunteersAndStudents.py [hunk 1]
```diff
@@ -3,8 +3,7 @@
 from flask import json, jsonify
 
 
-
-def searchVolunteers(query):
+def searchVolunteersAndStudents(query):
```
Comment (BrianRamsay) [inline app/logic/searchVolunteersAndStudents.py]: searchVolunteersAndStudents() is too wordy. It should reflect what you are actually doing in the function, not how it will be used. So, `searchStudents()`, or even `searchUsers()`.

Comment (BrianRamsay) [reply app/logic/searchVolunteersAndStudents.py]: Also, don't mess with JSON here. Let the controller handle changing the representation if it needs to. It's easier to use and easier to test if the function returns a dictionary.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
Right below the code you edited in routes.py is another search implementation. That one needs to be combined as well.

File: app/logic/searchVolunteersAndStudents.py (hunk 1 above)
Comment (sreynit02) [reply app/logic/searchVolunteersAndStudents.py]: changed `searchVolunteersAndStudents() ` to `searchUsers()`. Moved JSON to controller and changed test accordingly.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
There are Windows line endings (from development). I've fixed them in development so merge dev in here to resolve them in this branch.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
There is still a third search that should use searchUsers. See issue description.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
Getting close! But now events/routes.py has searchUsers and searchVolunteers, two different controller methods that are almost identical. I would probably change the front-ends to be able to use the same back-end search. It wouldn't surprise me if there's duplication in the javascript as well, to handle the results of a search.

**Comment by mupotsal** [comment]
We have updated the pull request. The search  methods are merged.

**Review by BrianRamsay — APPROVED** [review]
Nice work!
